STM32G473CBTx/Core/Inc/can.hpp

```diff
@@ -0,0 +1,115 @@
+#pragma once
+#include "main.h"
+
+struct CanDriver {
+
+    enum class FDCANOperatingMode : uint32_t {
+        Normal = FDCAN_MODE_NORMAL,
+        RestrictedOperation,
+        BusMonitoring,
+        InternalLoopback,
+        ExternalLoopback
+    };
+
+
+    struct Message {
+        static constexpr size_t MAX_DATA_LENGTH = 64;
```

Comment:
I had my facts mixed up slightly here. I should have used contexpr on it's own rather than static constexpr. I was under the impression that static constexpr were not stored in memory, but I was slightly incorect about that. In fact constexpr variables are the ones that are not stored in memory (as oposed to a regular const variable). They are compile time constants which can be type checked, but so long as a reference is not made to them (which you can't do with a define either) , they do not get included in the final binary. This is handy because it allows us to type check our constants while avoiding poluting the global context with constants